### ISSUE TITLE
Ensure EPD full upgrade case runs in a reasonable time on a few year old machine

### ISSUE THREAD

**Issue by cournape** [enthought/sat-solver#52]
Here, reasonable is defined as 2-3 seconds, and the scenario is the one in https://github.com/enthought/sat-solver/pull/51

**Comment by johntyree** [comment]
I currently get 3.5 seconds for `solve@dependency_solver.py` under cProfile.

**Comment by cournape** [comment]
Can you get the number for the pure solver (e.g. the call for `search`), but without running under cProfile. This is only for order of magnitude, so doing something like `with timeit(): run_solver...` is good enough.

**Comment by johntyree** [comment]
1.2 seconds under normal python execution on my mbp. Trying on old and slow VPS now as well.

**Comment by cournape** [comment]
OK. Once you have all the numbers, can you just create a small table in this issue w/ the different HW ? Just something we can refer to later.

In any case, the number are good enough for now.

**Comment by johntyree** [comment]
### MBP
![screen shot 2015-11-02 at 3 26 00 pm](https://cloud.githubusercontent.com/assets/108972/10895512/48cf6234-817a-11e5-9eda-bcaf50fc5adf.png)

Dreamhost VPS | Debian 6.0.10
----|----
CPU | AMD Opteron(tm) Processor 4122 @ 2GHz
MEM | 300MB

Create a 20MB noise file to work with,
```
dd if=/dev/urandom of=/tmp/noise bs=1024 count=20480
```
then the `bzip2` scenario is the running time of `bzip2 -9 /tmp/noise`

`solve` time computed by the following:
```
    # in DependencySolver
    def solve(self, request):
        """Given a request, computes the set of operations to apply to
        resolve it, or None if no resolution could be found.
        """
        start = now()
        rules = self._create_rules(request)
        sat_solver = MiniSATSolver.from_rules(rules, self._policy)
        solution = sat_solver.search()
        print(now() - start)
```

Scenario | MBP total | Dreamhost VPS total | MBP solve | Dreamhost VPS solve
---- | ---- | ---- | ---- | ----
20MB bzip2 | 5.3 s | 17.7 s | - | -
epd_full_upgrade | 3.7 | 13.91 | 2.5 | 9.42
iris | 0.98 | 3.2 | 0.38 | 1.4
simple_numpy | 0.31 | 0.84 | 0.0011 | 0.0045

**Comment by johntyree** [comment]
The VPS is apparently about a factor three slower than this macbook. Note that even *excluding* the entire call to `solve()`, we're taking longer than 2-3 seconds for `epd_full_upgrade`.

**Comment by cournape** [comment]
hm, 10 sec is a bit slower than I hoped. You checked that the VPS was not swapping (300 Mb is low !) ?

**Comment by cournape** [comment]
Still, even if a bit slow, l think it is acceptable for `0.1`.

**Comment by johntyree** [comment]
It's a very very slow machine and yes it has low mem. It's 5 bucks a month, what can I say :)

There is no swapping happening according to `/usr/bin/time`. I can't figure out a good way to separate out the time it takes for `enpkg` to download the index from the time it takes to do everything else, but if it helps we can also just look at `user` time for `enpkg --whats-new`:

MBP
```
3.66 real         2.46 user         0.26 sys
```

Dreamhost VPS
```
8.69user 0.52system 0:10.98elapsed 83%CPU (0avgtext+0avgdata 566880maxresident)k
376inputs+8216outputs (3major+57303minor)pagefaults 0swaps
```

Again about a factor 3 in `user`.

**Comment by johntyree** [comment]
This should have been closed by #68. Github needs better NLP.
